src/vim.js

```diff
@@ -868,8 +883,11 @@ export function initVim(CodeMirror) {
        * execute the bound command if a a key is matched. The function always
        * returns true.
        */
-      findKey: function(cm, key, origin) {
+      findKey: function(cm, rawKey, origin) {
         var vim = maybeInitVimState(cm);
+
+        let key = vim.expectLiteralNext ? rawKey : langmapRemapKey(rawKey);
```

Comment:
I think, If we move langmap logic to https://github.com/replit/codemirror-vim/blob/master/src/index.ts#L227C1-L228C1 we can remove the special case in replaymacro

Reply:
You can make remapKey a public api on vim, and in index.ts call that before  calling multiselectHandleKey

Reply:
Implemented on [this branch](https://github.com/Opisek/codemirror-vim/tree/expose-remap). Please check if that's how you pictured it, then I'll merge it into this PR.

Reply:
Exactly what i meant. That branch looks good to merge.